src/libostree/ostree-repo-pull.c

```diff
@@ -3257,9 +3265,27 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
           if (!ostree_repo_resolve_rev (pull_data->repo, ref, TRUE,
                                         &from_revision, error))
             goto out;
+
+          /* Determine whether the from revision we have is partial; this
+           * can happen if e.g. one uses `ostree pull --commit-metadata-only`.
+           * This mirrors the logic in get_best_static_delta_start_for().
+           */
+          if (from_revision)
+            {
+              OstreeRepoCommitState from_commitstate;
+
+              if (!ostree_repo_load_commit (pull_data->repo, from_revision, NULL,
+                                            &from_commitstate, error))
+                goto out;
+
+              /* Was it partial?  OK, we can't use it. */
+              if (commitstate_is_partial (pull_data, from_commitstate))
+                have_valid_from_commit = FALSE;
```

Comment:
That topic is kind of...messy.  Which is mostly papered over by the fact we aren't making (let's call them) "scratch deltas" in most streams I know of.  Under what scenarios is it better to download a scratch delta versus doing the archive-individual fetches?

A useful heuristic for the ostree-host case might be to only enable the "scratch delta" case if the repo is empty.  That'd fix the anaconda case.

This isn't conceptually a regression though I think.